utils/common/src/test/java/org/apache/brooklyn/util/exceptions/ExceptionsTest.java

```diff
@@ -59,6 +59,28 @@ public void testPropagateCheckedException() throws Exception {
     }
     
     @Test
+    public void testPropagateCheckedExceptionWithMessage() throws Exception {
+        String extraMsg = "my message";
+        Exception tothrow = new Exception("simulated");
+        try {
+            throw Exceptions.propagate(extraMsg, tothrow);
+        } catch (RuntimeException e) {
+            assertEquals(e.getMessage(), "my message");
+            assertEquals(e.getCause(), tothrow);
+        }
+    }
+    
+    @Test
+    public void testPropagateRuntimeExceptionIgnoresMessage() throws Exception {
+        NullPointerException tothrow = new NullPointerException("simulated");
+        try {
+            throw Exceptions.propagate("my message", tothrow);
```

Comment:
see `Asserts.shouldHaveFailedPreviously()` (useless in this simple case but i find it makes this type of test more readable)